src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java

```diff
@@ -407,15 +409,31 @@ static File resolvePath(@Nullable String path, File basedir) {
     return result;
   }
 
-  private List<File> mainSources(MavenProject pom)
-    throws MojoExecutionException {
+  private static void removeTarget(MavenProject pom, Collection<String> paths) {
+    paths.removeIf(pathStr -> {
+      Path path = Paths.get(pathStr);
+      Path relativePath;
+
+      if (path.isAbsolute()) {
+        Path baseDir = pom.getBasedir().toPath().toAbsolutePath();
+        relativePath = baseDir.relativize(path);
+      } else {
+        relativePath = path;
+      }
+
+      return relativePath.startsWith("target");
```

Comment:
Don't hardcode "target" folder since this is just the default value. You should use MavenProject.getBuild().getDirectory() (http://stackoverflow.com/questions/9318935/get-project-build-directory-from-mavenproject)